src/utility.c

```
@@ -0,0 +1,288 @@
+*/
+void Hash_1(CBigInt *h1, BYTE *msg, unsigned int len, CBigInt n)
+{
+	unsigned int ct, hlen,v, hlen_v,i,len2;
+	BYTE *msg2;
+	DWORD hash[8],Ha[10];
+	CBigInt HA,Mod;
+	ct = 1;              // Initialize counter to 1
+	v = 256;             // SM3 output is 256 bits
+	len2 = len + 1 + 4;   // H = Hv(0x01||Z||ct)
+	msg2 = (BYTE*)malloc(len);
+	memset(msg2,0,len2);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)